### DISCUSSION TITLE
New Plugins Query Model

### DISCUSSION THREAD

**Discussion by taooceros** [Flow-Launcher/Flow.Launcher#1103] [Ideas]
We can let plugins return a IEnumerable<Result> instead of a List<Result> (IAsyncEnumerable<Result> for Task<List<Result>>), which basically natively open the potential of some early result and some later results without invoking results again and again, but providing with what's new.
Further, I would like to investigate the potential of letting the listview to sort by itself so we won't need to readding to observable collection every time.

**Comment by jjw24** [comment]
Sounds promising 👍
